server/src/services/rbac.service.ts

```
@@ -0,0 +1,313 @@
+  }
+
+  /**
+   * Get resource information including owner
+   */
+  private async getResourceInfo(resourceType: ResourceType, resourceId: string): Promise<{ ownerId: string } | null> {
+    switch (resourceType) {
+      case ResourceType.LOAN:
+        const loan = await prisma.loan.findUnique({
+          where: { id: resourceId },
+          select: { borrowerId: true, lenderId: true }
+        });
```

Biome (2.4.9)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)